### DISCUSSION TITLE
Add a new C# operator "in" to check if an item is in the data structure.

### DISCUSSION THREAD

**Discussion by mckabue** [dotnet/csharplang#2635] [Language Ideas]
https://github.com/dotnet/roslyn/issues/37037#issue-464997182

**Comment by Tom01098** [comment]
So, where would this be used? Is it essentially an alias for a `Contains` method on the type? Is there an underlying interface for this behaviour?

**Comment by yaakov-h** [comment]
Duplicate of #1665.

**Comment by mckabue** [comment]
@Tom01098
Yes, operator `in` would be an alias to `Contains`.

We could however also use it with reflection to check if a property with a given name exists in an object, or class. [Just like `in` operator in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in)

**Comment by orthoxerox** [comment]
@McKabue reflection is something that should *not* be this convenient to use. It bypasses a lot of the safeguards built into C# and is much slower than regular programming. Its syntax is all "I understand the risks and want to proceed" by design.

**Comment by Tom01098** [comment]
@McKabue This would definitely never be implemented with reflection as it bypasses type safety.

If it's just an alias for `Contains`, then why not just use that? You're just saving a few characters and flipping the order; I know 'I don't like it' is a weak argument, but I think this ordering is less clear, especially as `in` doesn't really imply this behaviour.

**Comment by mckabue** [comment]
@Tom01098 I have nothing against `Contains`, I would just like an alternative. 
Also, I had a friend who was moving from `python` to `C#` and was asking why there is no `in` operator, generally to make it easy to transition from a language with similar syntax.

**Comment by john-h-k** [comment]
But why?

```cs
if (1 in someArr)
```

```cs
if (someArr.Contains(1))
```

It doesn't add much, it is just some syntactic sugar which is kinda ... useless. Why make an alternative for no reason? A proposal needs to be valuable to be accepted into the lang

**Comment by Tom01098** [comment]
> Also, I had a friend who was moving from `python` to `C#` and was asking why there is no `in` operator, generally to make it easy to transition from a language with similar syntax.

They're moving from a quite different language, C# isn't python v2 :)

Adding new ways of doing the same thing means people will have to learn both when reading and maintaining code, making it harder to do both.
